New page: aria-flowto (#10218)
mdn/content: files/en-us/web/accessibility/aria/attributes/aria-flowto/index.md

---
title: 'aria-flowto'
slug: Web/Accessibility/ARIA/Attributes/aria-flowto
tags: 
  - Accessibility
  - ARIA
  - ARIA attribute
  - ARIA property
  - aria-flowto
  - Reference
---

> The global `aria-flowto` attribute identifies the next element (or elements) in an alternate reading order of content which, at the user's discretion, allows assistive technology to override the general default of reading in document source order.

Reviewer: ```suggestion
The global `aria-flowto` attribute identifies the next element (or elements) in an alternate reading order of content. This allows assistive technology to override the general default of reading in document source order at the user's discretion.
```

## Description

> Web pages should be sequentially navigable. For this reason, developers are dissuaded from using the global {{htmlattrdef('tabindex')}} attribute which can change tab order and the CSS {{CSSXRef('order')}} property which can change visual order from the DOM order. However, sometimes a reading path different from the source order is required (like when `order` is use). For such cases, the `aria-flowto` attribute can make the content more accessible for users of assistive technologies. 

Reviewer: ```suggestion
Web pages should be sequentially navigable. For this reason, developers are dissuaded from using the global {{htmlattrdef('tabindex')}} attribute which can change tab order and the CSS {{CSSXRef('order')}} property, which can change visual order from the DOM order. However, in rare circumstances a reading path different from the source order is required. For such cases, the `aria-flowto` attribute can make the content more accessible for users of assistive technologies. 
```

Slight tweaks to stress this is something rare and to be avoided.

> The global `aria-flowto` attribute let's the developer indicate to assistive technology and their users which element or elements might want to be focused on next providing an alternate reading order to the source order. At the user's discretion, the attribute allows assistive technology to override the general default document source reading order.

Reviewer: ```suggestion
The global `aria-flowto` attribute lets the author indicate to assistive technology users which element or elements might want to be focused on next, providing an alternate reading order to the source order. This allows assistive technology to read a document in an order other than the default document source reading order.
```

The AT isn't overriding here, just consuming.

> When `aria-flowto` has a single {{htmlattrdef('id')}} reference, it allows assistive technologies to, at the user's request, go to the targeted object instead of reading the document in the order of the DOM. When the `aria-flowto` value is provided a space separated list of multiple `id` references, assistive technology should provide the user with a list of path choices with each `id` referenced being a choice. In supporting assitive technologies, the path choice names in determined by the name of each target element of the `aria-flowto` attribute. 

Reviewer: ```suggestion
When `aria-flowto` has a single {{htmlattrdef('id')}} reference, it allows assistive technologies to, at the user's request, go to the element targeted via that `id` instead of reading the document in the order of the DOM. When the `aria-flowto` value uses a space separated list of multiple `id` references, assistive technology can provide the user with a list of path choices, with each `id` referenced being a choice. The path choice names are determined by the accessible name of each target element of the `aria-flowto` attribute. 
```
